src/main/java/nextstep/subway/line/domain/Sections.java

```diff
@@ -0,0 +1,113 @@
+package nextstep.subway.line.domain;
+
+import nextstep.subway.station.domain.Station;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Embeddable;
+import javax.persistence.OneToMany;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Embeddable
+public class Sections {
+    private static final int MIN_SECTION_SIZE = 1;
+
+    @OneToMany(mappedBy = "line", cascade = {CascadeType.PERSIST, CascadeType.MERGE}, orphanRemoval = true)
+    private List<Section> sections = new ArrayList<>();
+
+    public Sections() {
+    }
+
+    public void add(Section newSection) {
+        if (sections.isEmpty()) {
+            sections.add(newSection);
+            return;
+        }
+        addSection(newSection);
+    }
+
+    private void addSection(Section newSection) {
+        Section section = sections.stream()
+                .filter(oldSection -> oldSection.isNotDuplicate(newSection))
+                .findFirst()
+                .map(oldSection -> oldSection.divide(newSection))
+                .orElseThrow(() -> new IllegalArgumentException("구간을 연결할 상행역 또는 하행역이 존재해야 합니다."));
+
+        sections.add(section);
+    }
+
+    public List<Section> getSections() {
+        List<Section> sortedSections = new ArrayList<>();
+        Section section = findFirstSection();
+        sortedSections.add(section);
+
+        while (sortedSections.size() < this.sections.size()) {
+            section = findNextStation(section.getDownStation());
+            sortedSections.add(section);
+        }
+
+        return Collections.unmodifiableList(sortedSections);
+    }
+
+    public Section findFirstSection() {
+        List<Station> downStations = getDownStations();
+
+        return sections.stream()
+                .filter(section -> !downStations.contains(section.getUpStation()))
+                .findFirst()
+                .orElseThrow(() -> new RuntimeException("상행 종점역을 찾을 수 없습니다."));
+    }
+
+    private Section findNextStation(Station downStation) {
+        return sections.stream()
+                .filter(section -> section.getUpStation().equals(downStation))
+                .findFirst()
+                .orElseThrow(() -> new RuntimeException("이어지는 구간을 찾을 수 없습니다."));
+    }
+
+    public List<Station> getDownStations() {
+        return sections.stream()
+                .map(Section::getDownStation)
+                .collect(Collectors.toList());
+    }
+
+    public boolean isEmpty() {
+        return sections.isEmpty();
+    }
+
+    public void remove(Station station) {
+        if (sections.size() == MIN_SECTION_SIZE) {
+            throw new IllegalArgumentException("노선의 구간이" + MIN_SECTION_SIZE + "개 이하인 경우 구간을 삭제할 수 없습니다.");
+        }
```

Comment:
예외 메시지와 같이 이하로 비교해주시는 것이 혹시 모를 에러 상황을 방지할 수 있을 것 같아요 :)
```suggestion
        if (sections.size() <= MIN_SECTION_SIZE) {
            throw new IllegalArgumentException("노선의 구간이" + MIN_SECTION_SIZE + "개 이하인 경우 구간을 삭제할 수 없습니다.");
        }
```

Reply:
넵 알겠습니다 :)